ejb/WFLY-13354-galleon-for-ejb.adoc

```diff
@@ -0,0 +1,110 @@
+= Galleon layers for the EJB subsystem
+:author:            Brian Stansberry; Yeray Borges; Cheng Fang
+:email:             cfang@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The goal of this feature is to supply a list of https://docs.wildfly.org/galleon/#_layers[Galleon layers]
+for the EJB subsystem to support common predefined EJB feature sets.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-13354[WFLY-13354 Add Galleon layers for the EJB subsystem]
+* https://issues.redhat.com/browse/EAP7-1444[EAP7-1444 Galleon Layers in EAP XP 2]
+
+=== Related Issues
+* https://issues.redhat.com/browse/WFLY-13433[Improve capability support in EJB3 subsystem]
+* https://issues.redhat.com/browse/WFLY-13356[Add a Galleon layer for Webservices]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+These analysis and requirements are based on https://github.com/bstansberry/wildfly/commits/ejb-layers2[Brian's proposal].
+
+If a user is trying to slim their server I think of four key things they want to do in descending order of importance:
+
+1. Eliminate unnecessary open ports.
+
+2. Eliminate unnecessary exposure.
+
+3. Eliminate jars, both to save filesystem footprint and to reduce any theoretical attack surface.
+
+4. Reduce configuration clutter and unnecessary services.
+
+Based on the above factors, the following layers are to be supported:
+
+1. ejb3-lite. This is like what we provide in standalone.xml, but without the MDB instance pool or the remote connector resource.
+(A webservices layer would depend on this.)
+
+2. ejb3. Builds (i.e. depends) on ejb3-lite and adds the discovery subsystem, the MDB pool and the remote connector resource.
+This looks what we provide in standalone-full.xml, except no IIOP integration.
+
+3. ejb3-iiop. Builds (i.e. depends) on *ejb3-lite* and adds the iiop subsystem plus the integration resource in the EJB3 subsystem.
+This *could* depend on *ejb3* instead, but that means EJB is unnecessarily exposed over the HTTP interface and a lot of
+dependencies are brought in for messaging integration.
+
+A user could provision ejb3 + ejb3-iiop and get the equivalent to what's in standalone-full.xml.
```

Comment:
We won't be supporting the legacy security subsystem in bootable jar so this equivalence may not be possible.

Instead some consideration to the default security domain may be needed.